### PR TITLE
PDK convert 1.3.2

### PR DESCRIPTION
**Pull request by HelenCampbell**

### CODE REVIEW

**Comment by HelenCampbell** [comment]
Hey @glennsarti , This is the first module for cross-platform we've converted to PDK. Would you do me a solid and review this from a Windows standpoint? Just before we go ahead and convert more Windows supported modules we want to make sure we haven't missed anything. Thanks!

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,23 @@
-# This file is generated by ModuleSync, do not edit.
-*.iml
 .*.sw[op]
-.DS_Store
```
Comment (glennsarti) [inline .gitignore]: DS_Store is no longer in gitignore.  Is this intentional?

File: appveyor.yml [hunk 2]
```diff
@@ -3,39 +3,61 @@ skip_commits:
...
-- 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
-- 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
-- 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
+  - SET
+  - 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
+  - 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
+  - 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+  - 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
 environment:
+  PUPPET_GEM_VERSION: ~> 4.0
```
Comment (glennsarti) [inline appveyor.yml]: I'm confused how this got into the environment.  It's not specified in the sync.yml file

File: appveyor.yml [hunk 3]
```diff
@@ -3,39 +3,61 @@ skip_commits:
...
+      RUBY_VERSION: 24-x64
+      CHECK: rubocop
+    -
+      RUBY_VERSION: 24-x64
+      CHECK: spec
+    -
+      RUBY_VERSION: 21-x64
+      CHECK: spec
+    -
+      PUPPET_GEM_VERSION: ~> 5.0
```
Comment (glennsarti) [inline appveyor.yml]: I'm confused how this got into the environment.  It's not specified in the sync.yml file

File: appveyor.yml [hunk 4]
```diff
@@ -3,39 +3,61 @@ skip_commits:
...
-  - PUPPET_GEM_VERSION: ~> 4.0
-    RUBY_VER: 21-x64
-  - PUPPET_GEM_VERSION: ~> 4.0
-    RUBY_VER: 23
-  - PUPPET_GEM_VERSION: ~> 4.0
-    RUBY_VER: 23-x64
+    -
+      RUBY_VERSION: 24-x64
+      CHECK: syntax lint
+    -
```
Comment (glennsarti) [inline appveyor.yml]: Looks like a formatting eror here,  Shouldn't be blank

File: appveyor.yml [hunk 5]
```diff
@@ -3,39 +3,61 @@ skip_commits:
...
-    ruby -ropenssl -e 'puts \"OpenSSL Version - #{OpenSSL::OPENSSL_VERSION}\"; puts \"OpenSSL Library Version - #{OpenSSL::OPENSSL_LIBRARY_VERSION}\"'
-- bundle install --jobs 4 --retry 2 --without system_tests
-- type Gemfile.lock
+  - set PATH=C:\Ruby%RUBY_VERSION%\bin;%PATH%
+  # Due to a bug in the version of OpenSSL shipped with Ruby 2.4.1 on Windows
+  # (https://bugs.ruby-lang.org/issues/11033). Errors are ignored because the
+  # mingw gem calls out to pacman to install OpenSSL which is already
+  # installed, causing gem to raise a warning that powershell determines to be
+  # a fatal error.
+  - ps: |
```
[the remark this answers is not in the document]
Comment (glennsarti) [reply appveyor.yml]: Requesting changes on this PR for this

**Comment by glennsarti** [comment]
Also in Travis

Ruby 2.1.9  PUPPET_GEM_VERSION~>4.0 CHECK=spec is being run twice.

File: .gitignore (hunk 1 above)
Comment (HelenCampbell) [reply .gitignore]: Unintentional - Good catch - PR for fix to template https://github.com/puppetlabs/pdk-templates/pull/15
I'll fix this in the PR also.

File: appveyor.yml (hunk 2 above)
Comment (HelenCampbell) [reply appveyor.yml]: This comes from the config_defaults specified in the templates: https://github.com/puppetlabs/pdk-templates/blob/master/config_defaults.yml#L31

File: appveyor.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (HelenCampbell) [reply appveyor.yml]: Another good catch - And another template PR for fix: https://github.com/puppetlabs/pdk-templates/pull/16
I'll fix this in the PR before re-requesting review.

**Comment by HelenCampbell** [comment]
Hey @glennsarti , sorry for the slow feedback loop, but this is ready for re-review :) I have appveyor in a better state now along with a few other tweaks of your recommendation.

File: appveyor.yml [hunk 6]
```diff
@@ -3,39 +3,61 @@ skip_commits:
...
+  - 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+  - 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
 environment:
+  PUPPET_GEM_VERSION: ~> 4.0
   matrix:
-  - PUPPET_GEM_VERSION: ~> 4.0
-    RUBY_VER: 21
-  - PUPPET_GEM_VERSION: ~> 4.0
-    RUBY_VER: 21-x64
-  - PUPPET_GEM_VERSION: ~> 4.0
```
[the remark this answers is not in the document]
Comment (glennsarti) [reply appveyor.yml]: You want to end up with a configuration similar to;

```
  - PUPPET_GEM_VERSION: ~> 4.0
    RUBY_VER: 21
  - PUPPET_GEM_VERSION: ~> 4.0
    RUBY_VER: 21-x64
  - PUPPET_GEM_VERSION: ~> 5.0
    RUBY_VER: 24
  - PUPPET_GEM_VERSION: ~> 5.0
    RUBY_VER: 24-x64
```

Unfortunately it's testing 2.4 and 2.3 with Puppet 4.0

File: .travis.yml [hunk 7]
```diff
@@ -1,31 +1,58 @@
...
+  - gem --version
+  - bundle -v
+script:
+  - 'bundle exec rake $CHECK'
+bundler_args: --without system_tests
+rvm:
+  - 2.4.1
+  - 2.1.9
+env:
+  - PUPPET_GEM_VERSION="~> 4.0" CHECK=spec
```
Comment (glennsarti) [inline .travis.yml]: This is testing Puppet 4.0 with 2.4.1 and PUppet 5.0 with 2.1.9.  This seems a waste.

Note Puppet 4 with 2.1.9 is still being tested twice

**Comment by HelenCampbell** [comment]
@glennsarti I think it's in solid shape now, a quick re-review if you would?

File: .travis.yml [hunk 8]
```diff
@@ -1,31 +1,54 @@
-#This file is generated by ModuleSync, do not edit.
 ---
 sudo: false
+dist: trusty
 language: ruby
 cache: bundler
-script: "bundle exec rake release_checks"
+before_install:
+  - bundle -v
+  - rm Gemfile.lock || true
```
Comment (glennsarti) [inline .travis.yml]: Huh...that's new.  Doesn't need to be changed, just an observation.

File: appveyor.yml [hunk 9]
```diff
@@ -3,39 +3,48 @@ skip_commits:
...
+matrix: 
   fast_finish: true
 install:
-- SET PATH=C:\Ruby%RUBY_VER%\bin;%PATH%
-- ps: |
-    gem list openssl
-    ruby -ropenssl -e 'puts \"OpenSSL Version - #{OpenSSL::OPENSSL_VERSION}\"; puts \"OpenSSL Library Version - #{OpenSSL::OPENSSL_LIBRARY_VERSION}\"'
-- bundle install --jobs 4 --retry 2 --without system_tests
-- type Gemfile.lock
+  - set PATH=C:\Ruby%RUBY_VERSION%\bin;%PATH%
```
Comment (glennsarti) [inline appveyor.yml]: I notice in travis Gemfile is deleted, but not in Appveyor.  Is there a reason for the difference?

Comment (HelenCampbell) [reply appveyor.yml]: Not that I'm aware of to be honest, the removal of the Gemfile.lock seems to be coming in from the pdk template. Looks fairly harmless so I'll leave it in.

**Pull request merged**